src/common/Language/Java/Inline/Internal.hs

```diff
@@ -202,9 +199,11 @@ blockQQ config input = do
              $(return $ foldr (\a b -> TH.TupE [TH.VarE a, b]) (TH.TupE []) thnames)
              Proxy
              (\ $(return $ foldr (\a b -> TH.TupP [TH.VarP a, b]) (TH.TupP []) thnames') ->
-               $(TH.varE (qqCallStatic config))
-               (fromString $(TH.stringE ("io.tweag.inlinejava." ++ mangle thismod)))
-               (fromString $(TH.stringE mname))
-               $(TH.listE args)
+                $(TH.appsE
+                    ([ TH.varE (qqCallStatic config)
+                     , [| fromString $(TH.stringE ("io.tweag.inlinejava." ++ mangle thismod)) |]
+                     , [| fromString $(TH.stringE mname) |]
+                     ] ++ map TH.varE thnames')
+                 )
```

Comment:
This code is used both by the linear and the non-linear interfaces. Looks like we will have to abstract the way in which arguments are passed.